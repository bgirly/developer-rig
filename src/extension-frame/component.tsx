import * as React from 'react';
import './component.sass';
import { ExtensionPlatform, ExtensionViewType} from '../constants/extension-coordinator';

const IFRAME_CLASS = 'extension-frame';
const EXTENSION_FRAME_INIT_ACTION = 'extension-frame-init';

export interface PublicProps {
  className: string;
  frameId: string;
  extension: ExtensionCoordinator.ExtensionObject;
  installationAbilities: ExtensionCoordinator.ExtensionInstallationAbilities;
  type: string;
  mode: string;
  bindIframeToParent: (iframe: HTMLIFrameElement) => void;
}

type Props = PublicProps;

export class ExtensionFrame extends React.Component<Props> {
  public iframe: HTMLIFrameElement;

  public componentDidMount() {
    if (this.iframe) {
      this.iframe.onload = this.extensionFrameInit;
    }
  }

  public render() {
    return (
      <iframe
        ref={this.bindIframeRef}
        src={process.env.PUBLIC_URL + '/extension-frame.html'}
        frameBorder={0}
        className={`rig-frame ${IFRAME_CLASS}`}
        title={this.props.frameId}
      />
    );
  }

  private bindIframeRef = (iframe: HTMLIFrameElement) => {
    this.iframe = iframe;
    this.props.bindIframeToParent(iframe);
  }

  public extensionFrameInit = () => {
    const extensionFrameParams: ExtensionCoordinator.ExtensionFrameParams = {
      anchor: this.props.type as ExtensionCoordinator.ExtensionAnchor,
      channelId: parseInt(process.env.EXT_CHANNEL_ID, 0),
      extension: this.props.extension,
      iframeClassName: IFRAME_CLASS,
      installationAbilities: this.props.installationAbilities,
      loginId: null,
      mode: this.props.mode as ExtensionCoordinator.ExtensionMode,
      platform: (this.props.type === ExtensionViewType.Mobile) ? ExtensionPlatform.Mobile : ExtensionPlatform.Web,
      trackingProperties: {},
    }

    const data = {
      action: EXTENSION_FRAME_INIT_ACTION,
      extension: extensionFrameParams,
      channelId: process.env.EXT_CHANNEL_ID,
      frameId: this.props.frameId,
    };

    this.iframe.contentWindow.postMessage(data, '*');
  }
}
