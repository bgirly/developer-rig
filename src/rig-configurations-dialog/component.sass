@import "../dialog-component/component.sass"
.rig-configurations-view


.rig-configurations-view__background
  @extend .dialog__backdrop

.rig-configurations-view__dialog
  @extend .dialog
  height: 400px

.rig-configurations-view__content
  @extend .container__left-block
  white-space: pre
  overflow-y: auto
  height: 250px
  width: 95%

.dialog__top-bar-container
  @extend .top-bar

.top-bar-container__title
  @extend .top-bar-element-left
  margin-top: 5px

.top-bar-container__escape
  @extend .top-bar-element-right
  width: $top-bar-dimension
  cursor: pointer
  width: 4em
  margin-top: 5px

.dialog__divider
  @extend .dialog__divider-width

.bottom-bar__refresh
  @include bottom-bar-box(white, #7D55C7)

.bottom-bar__cancel
  @include bottom-bar-box(#7D55C7, white)
