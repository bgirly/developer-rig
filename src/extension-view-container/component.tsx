import * as React from 'react';
import { ExtensionView } from '../extension-view';
import { ExtensionViewButton } from '../extension-view-button';
import { ExtensionMode } from '../constants/extension-coordinator';
import './component.sass';
import { RigExtensionView } from '../core/models/rig';

interface ExtensionViewContainerProps {
  extensionViews: RigExtensionView[];
  openEditViewHandler?: (id: string) => void;
  deleteExtensionViewHandler: (id: string) => void;
  openExtensionViewHandler: Function;
}

type Props = ExtensionViewContainerProps;

const ConfigNames: { [key: string]: string; } = {
  [ExtensionMode.Config]: 'Broadcaster Configuration',
  [ExtensionMode.Dashboard]: 'Broadcaster Live Dashboard',
};

export class ExtensionViewContainer extends React.Component<Props> {
  private openExtensionViewDialog = () => {
    this.props.openExtensionViewHandler();
  }

  public render() {
    let extensionViews: JSX.Element[] = [];
    if (this.props.extensionViews && this.props.extensionViews.length > 0) {
      extensionViews = this.props.extensionViews.map((view) => ((
        <ExtensionView
          key={view.id}
          id={view.id}
          extension={view.extension}
          installationAbilities={view.features}
          type={view.type}
          mode={view.mode}
          role={view.mode === ExtensionMode.Viewer ? view.role : ConfigNames[view.mode]}
          frameSize={view.frameSize}
          position={{ x: view.x, y: view.y }}
          linked={view.linked}
          orientation={view.orientation}
          openEditViewHandler={this.props.openEditViewHandler}
          deleteViewHandler={this.props.deleteExtensionViewHandler}
        />
      )));
    }

    return (
      <div className='view-container-wrapper'>
        <div className="view-container">
          {extensionViews}
        </div>
        <div>
          <ExtensionViewButton
            onClick={this.openExtensionViewDialog}>
          </ExtensionViewButton>
        </div>
      </div>
    );
  }
}
