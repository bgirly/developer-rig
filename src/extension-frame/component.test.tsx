import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { setupMountTest } from '../tests/enzyme-util/mount';
import { createExtensionForTest } from '../tests/constants/extension';
import { ExtensionFrame } from './component';
import { ExtensionViewType, ExtensionAnchor, ExtensionMode } from '../constants/extension-coordinator';

const defaultPropGenerator = () => ({
  className: 'view',
  frameId: '0',
  installationAbilities: {
    isChatEnabled: true,
  },
  extension: createExtensionForTest(),
  type: ExtensionAnchor.Panel,
  mode: ExtensionMode.Viewer,
  iframe: '',
  bindIframeToParent: jest.fn(),
});

const setupShallow = setupShallowTest(ExtensionFrame, defaultPropGenerator);
const setupMount = setupMountTest(ExtensionFrame, defaultPropGenerator);

describe('<ExtensionFrame />', () => {
  it('onload postMessages data correctly', () => {
    const { wrapper } = setupMount();

    const mockIframeRef: any = {
      contentWindow: {
        postMessage: jest.fn(),
      },
    };

    const instance = wrapper.instance() as ExtensionFrame;
    instance.iframe = mockIframeRef;
    instance.extensionFrameInit();
    expect(mockIframeRef.contentWindow.postMessage).toHaveBeenCalledWith({
      action: 'extension-frame-init',
      channelId: process.env.EXT_CHANNEL_ID,
      extension: {
        anchor: 'panel',
        channelId: NaN,
        extension: {
          authorName: 'test',
          clientId: 'mockClientId',
          bitsEnabled: false,
          description: 'description',
          iconUrl: 'icon_url',
          id: 'id',
          name: 'name',
          requestIdentityLink: false,
          sku: 'sku',
          summary: 'summary',
          token: 'token',
          vendorCode: 'vendorCode',
          version: '0.1',
          state: 'Released',
          views: {
            config: {
              canLinkExternalContent: false,
              viewerUrl: 'test',
            },
            component: {
              aspectHeight: 3000,
              aspectWidth: 2500,
              canLinkExternalContent: false,
              viewerUrl: 'test',
              zoom: false,
              zoomPixels: 100,
            },
            liveConfig: {
              canLinkExternalContent: false,
              viewerUrl: 'test',
            },
            mobile: {
              viewerUrl: 'test',
            },
            panel: {
              canLinkExternalContent: false,
              height: 300,
              viewerUrl: 'test',
            },
            videoOverlay: {
              canLinkExternalContent: false,
              viewerUrl: 'test',
            },
          },
          whitelistedConfigUrls: ['foo'],
          whitelistedPanelUrls: ['bar'],
        },
        iframeClassName: 'extension-frame',
        installationAbilities: {
          isChatEnabled: true,
        },
        loginId: null,
        mode: 'viewer',
        platform: 'web',
        trackingProperties: {}
      },
      frameId: '0'
    }, '*');
  });

  it('onload postMessages data correctly when platform is mobile', () => {
    const { wrapper } = setupMount({
      type: ExtensionViewType.Mobile
    });

    const mockIframeRef: any = {
      contentWindow: {
        postMessage: jest.fn(),
      },
    };

    const instance = wrapper.instance() as ExtensionFrame;
    instance.iframe = mockIframeRef;
    instance.extensionFrameInit();
    expect(mockIframeRef.contentWindow.postMessage).toHaveBeenCalledWith({
      action: 'extension-frame-init',
      channelId: process.env.EXT_CHANNEL_ID,
      extension: {
        anchor: 'mobile',
        channelId: NaN,
        extension: {
          authorName: 'test',
          clientId: 'mockClientId',
          bitsEnabled: false,
          description: 'description',
          iconUrl: 'icon_url',
          id: 'id',
          name: 'name',
          requestIdentityLink: false,
          sku: 'sku',
          summary: 'summary',
          token: 'token',
          vendorCode: 'vendorCode',
          version: '0.1',
          state: 'Released',
          views: {
            config: {
              canLinkExternalContent: false,
              viewerUrl: 'test',
            },
            component: {
              aspectHeight: 3000,
              aspectWidth: 2500,
              canLinkExternalContent: false,
              viewerUrl: 'test',
              zoom: false,
              zoomPixels: 100,
            },
            liveConfig: {
              canLinkExternalContent: false,
              viewerUrl: 'test',
            },
            mobile: {
              viewerUrl: 'test',
            },
            panel: {
              canLinkExternalContent: false,
              height: 300,
              viewerUrl: 'test',
            },
            videoOverlay: {
              canLinkExternalContent: false,
              viewerUrl: 'test',
            },
          },
          whitelistedConfigUrls: ['foo'],
          whitelistedPanelUrls: ['bar'],
        },
        iframeClassName: 'extension-frame',
        installationAbilities: {
          isChatEnabled: true,
        },
        loginId: null,
        mode: 'viewer',
        platform: 'mobile',
        trackingProperties: {}
      },
      frameId: '0'
    }, '*');
  });

  describe('when in live config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionViewType.LiveConfig,
        mode: ExtensionMode.Dashboard,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in live config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionViewType.LiveConfig,
        mode: ExtensionMode.Dashboard,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionViewType.Config,
        mode: ExtensionMode.Config,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in panel mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionAnchor.Panel,
        mode: ExtensionMode.Viewer,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in video overlay mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionAnchor.Overlay,
        mode: ExtensionMode.Viewer,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when in video overlay mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        type: ExtensionAnchor.Overlay,
        mode: ExtensionMode.Viewer,
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
