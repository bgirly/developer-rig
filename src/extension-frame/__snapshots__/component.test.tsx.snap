// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtensionFrame /> when in config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    className="view"
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "test",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "mobile": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "test",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    mode="config"
    type="config"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in live config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    className="view"
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "test",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "mobile": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "test",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    mode="dashboard"
    type="liveConfig"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in live config mode renders correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    className="view"
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "test",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "mobile": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "test",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    mode="dashboard"
    type="liveConfig"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in panel mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    className="view"
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "test",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "mobile": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "test",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    mode="viewer"
    type="panel"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in video overlay mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    className="view"
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "test",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "mobile": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "test",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    mode="viewer"
    type="video_overlay"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionFrame /> when in video overlay mode renders correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionFrame
    bindIframeToParent={[Function]}
    className="view"
    extension={
        Object {
            "authorName": "test",
            "bitsEnabled": false,
            "clientId": "mockClientId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentityLink": false,
            "sku": "sku",
            "state": "Released",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "viewerUrl": "test",
                "zoom": false,
                "zoomPixels": 100,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
              "mobile": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "test",
              },
              "videoOverlay": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    frameId="0"
    iframe=""
    installationAbilities={
        Object {
            "isChatEnabled": true,
          }
    }
    mode="viewer"
    type="video_overlay"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "rig-frame extension-frame",
      "frameBorder": 0,
      "src": "/extension-frame.html",
      "title": "0",
    },
    "ref": [Function],
    "rendered": null,
    "type": "iframe",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "rig-frame extension-frame",
        "frameBorder": 0,
        "src": "/extension-frame.html",
        "title": "0",
      },
      "ref": [Function],
      "rendered": null,
      "type": "iframe",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
